### ISSUE TITLE
WinForms Designer does not correctly round-trip Double values

### ISSUE THREAD

**Issue by DustinCampbell** [dotnet/roslyn#4149]
Repro Steps:

1. Create a new WinForms application (C# or VB -- doesn't matter).
2. Create a new UserControl with a public property of type double.
3. Build.
4. Drag UserControl1 from the Toolbox and drop onto Form1.
5. Set the double property to 9.2233720368547758E+18.
6. Open Form1.Designer.cs and verify that the double property is set to 9.2233720368547758E+18 in InitializeComponent().
7. Make a change to InitializeComponent(). Any change is OK -- even whitespace.
8. Go back to the Form Designer and change a property.
9. Check the Form1.Designer.cs file again and you'll see that the double property is now set to 9.22337203685478E+18.

**Comment by gafter** [comment]
@jaredpar @DustinCampbell 

The following one-line program fails when compiled with the Roslyn command-line compiler due to the underlying bug.

```cs
        System.Diagnostics.Debug.Assert(0.6822871999174 == double.Parse("0.6822871999174"));
```

An incorrect double value is baked into the generated assembly for the left-hand argument of ==. I suspect this may be due to some broken version of the CLR library against which the command-line compiler is run, but I have not diagnosed it further.

**Comment by gafter** [comment]
See #4221 for the related compiler issue.

**Comment by DustinCampbell** [comment]
@gafter This isn't related to #4221. This particular issue is purely with the implementation of IDE's serialization contact with the WinForms Designer. I already have a fix for this one. I'm just writing more test coverage.

**Comment by DustinCampbell** [comment]
Fixed with PR #4333.
